crypto/mem_sec.c

```diff
@@ -20,7 +20,9 @@
 
 #include <string.h>
 
-#if defined(OPENSSL_SYS_LINUX) || defined(OPENSSL_SYS_UNIX)
+/* e_os.h includes unistd.h, which defines _POSIX_VERSION */
+#if defined(OPENSSL_SYS_UNIX) \
+    && defined(_POSIX_VERSION) && _POSIX_VERSION >= 200112L
```

Comment:
If anyone wonders why I removed `defined(OPENSSL_SYS_LINUX)`, it's because `OPENSSL_SYS_UNIX` is defined anyway on that platform...

Reply:
Well, formally speaking one can as well omit defined(OPENSSL_SYS_UNIX). As we already established it's rather a notion of POSIX compliance, and _POSIX_VERSION describes it better...

Reply:
But this merge request is probably not right place to discuss this kind of thing. Approving...